src/sentry/utils/kafka_config.py

```diff
@@ -0,0 +1,75 @@
+from __future__ import absolute_import
+import six
+from django.conf import settings
+
+SUPPORTED_KAFKA_CONFIGURATION = (
+    # Check https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md
+    # for the full list of available options
+    "bootstrap.servers",
+    "compression.type",
+    "message.max.bytes",
+    "sasl.mechanism",
+    "sasl.username",
+    "sasl.password",
+    "security.protocol",
+)
+COMMON_SECTION = "common"
+PRODUCERS_SECTION = "producers"
+CONSUMERS_SECTION = "consumers"
+ADMIN_SECTION = "admin"
+KNOWN_SECTIONS = (COMMON_SECTION, PRODUCERS_SECTION, CONSUMERS_SECTION, ADMIN_SECTION)
+
+
+def _get_legacy_kafka_cluster_options(cluster_name):
+    options = settings.KAFKA_CLUSTERS[cluster_name]
+
+    options = {k: v for k, v in options.items() if k not in KNOWN_SECTIONS}
+    if "bootstrap.servers" in options:
+        if isinstance(options["bootstrap.servers"], (list, tuple)):
+            options["bootstrap.servers"] = ",".join(options["bootstrap.servers"])
+    return options
+
+
+def _get_kafka_cluster_options(
+    cluster_name, config_section, with_legacy=False, override_params=None
+):
+    options = {}
+    custom_options = settings.KAFKA_CLUSTERS[cluster_name].get(config_section, {})
+    common_options = settings.KAFKA_CLUSTERS[cluster_name].get(COMMON_SECTION, {})
+    legacy_options = _get_legacy_kafka_cluster_options(cluster_name)
+    if with_legacy and legacy_options:
+        # we prefer these ones
+        options.update(legacy_options)
+    else:
+        options.update(common_options)
+        options.update(custom_options)
+        # legacy override of bootstrap.servers should be preserved
+        if "bootstrap.servers" in legacy_options:
+            options["bootstrap.servers"] = legacy_options["bootstrap.servers"]
```

Comment:
Thanks for noticing this, but I don't think this is the right fix. If there is a bootstrap_server parameter I think we should ignore the custom fields just like we do in the branch above. 
Not doing so means we would have the consumer and the producer configuration behave differently: For producers, the mere presence of legacy fields will make us ignore new fields. For consumers, we would always pick the new field and override a specific one with the legacy config.

I would make them behave the same. So if there is a bootstrap.servers legacy field ignore the new fields in the consumer as well.

Reply:
you are right again 🤦 . pushing the fix

Reply:
updated